### PR TITLE
Explain a little bit on how to prepare Dockerfile before building kube-build image, to avoid confusion. 

### PR DESCRIPTION
**Pull request by zzxwill**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Explain a little bit on how to prepare Dockerfile before building kube-build image, to avoid confusion. 

**Which issue(s) this PR fixes**:

Fixes 89363

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
```

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ The following scripts are found in the [`build/`](.) directory. Note that all sc
 
 ## Basic Flow
 
-The scripts directly under [`build/`](.) are used to build and test.  They will ensure that the `kube-build` Docker image is built (based on [`build/build-image/Dockerfile`](build-image/Dockerfile)) and then execute the appropriate command in that container.  These scripts will both ensure that the right data is cached from run to run for incremental builds and will copy the results back out of the container.
+The scripts directly under [`build/`](.) are used to build and test.  They will ensure that the `kube-build` Docker image is built (based on [`build/build-image/Dockerfile`](build-image/Dockerfile) and after base image's `KUBE_BUILD_IMAGE_CROSS_TAG` from Dockerfile is repalced with one of those actual tags of the base image, like `v1.13.9-2`)) and then execute the appropriate command in that container.  These scripts will both ensure that the right data is cached from run to run for incremental builds and will copy the results back out of the container.
```
Comment (fejta) [inline build/README.md]: replaced

**Review by fejta — COMMENTED** [review]
/ok-to-test

File: build/README.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ The following scripts are found in the [`build/`](.) directory. Note that all sc
 
 ## Basic Flow
 
-The scripts directly under [`build/`](.) are used to build and test.  They will ensure that the `kube-build` Docker image is built (based on [`build/build-image/Dockerfile`](build-image/Dockerfile)) and then execute the appropriate command in that container.  These scripts will both ensure that the right data is cached from run to run for incremental builds and will copy the results back out of the container.
+The scripts directly under [`build/`](.) are used to build and test.  They will ensure that the `kube-build` Docker image is built (based on [`build/build-image/Dockerfile`](build-image/Dockerfile) and after base image's `KUBE_BUILD_IMAGE_CROSS_TAG` from Dockerfile is repalced with one of those actual tags of the base image, like `v1.13.9-2`) and then execute the appropriate command in that container.  These scripts will both ensure that the right data is cached from run to run for incremental builds and will copy the results back out of the container.
```
Comment (fejta) [inline build/README.md]: replaced (still misspelled)

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

**Comment by spiffxp** [comment]
@zzxwill you'll need to rebase to remove the merge commit from this PR, see https://github.com/kubernetes/kubernetes/pull/89364#issuecomment-602554843

**Pull request merged**